index.js

```diff
@@ -278,27 +280,55 @@ class GithubScm extends Scm {
             "else echo 'sd-step exec core/git'; fi)";
         const command = [];
 
-        // Git clone
-        command.push(`echo Cloning ${checkoutUrl}, on branch ${config.branch}`);
+        // Export environment variables
+        command.push('echo Exporting environment variables');
         command.push('if [ ! -z $SCM_CLONE_TYPE ] && [ $SCM_CLONE_TYPE = ssh ]; ' +
             `then export SCM_URL=${sshCheckoutUrl}; ` +
             'elif [ ! -z $SCM_USERNAME ] && [ ! -z $SCM_ACCESS_TOKEN ]; ' +
             `then export SCM_URL=https://$SCM_USERNAME:$SCM_ACCESS_TOKEN@${checkoutUrl}; ` +
             `else export SCM_URL=https://${checkoutUrl}; fi`);
-        command.push(`${gitWrapper} `
-                + `"git clone --recursive --quiet --progress --branch ${config.branch} `
-                + '$SCM_URL $SD_SOURCE_DIR"');
-        // Reset to SHA
-        command.push(`${gitWrapper} "git reset --hard ${checkoutRef} --"`);
-        command.push(`echo Reset to ${checkoutRef}`);
-        // Set config
-        command.push('echo Setting user name and user email');
-        command.push(`${gitWrapper} "git config user.name ${this.config.username}"`);
-        command.push(`${gitWrapper} "git config user.email ${this.config.email}"`);
         command.push('export GIT_URL=$SCM_URL.git');
         // git 1.7.1 doesn't support --no-edit with merge, this should do same thing
         command.push('export GIT_MERGE_AUTOEDIT=no');
 
+        // Set config
+        command.push('echo Setting user name and user email');
+        command.push(`${gitWrapper} "git config --global user.name ${this.config.username}"`);
+        command.push(`${gitWrapper} "git config --global user.email ${this.config.email}"`);
+
+        if (config.manifest) {
+            const repoDownloadUrl = 'https://storage.googleapis.com/git-repo-downloads/repo';
+            const sdRepoReleasesUrl = 'https://github.com/screwdriver-cd/sd-repo/releases/latest';
+
+            command.push('echo Checking out code using the repo manifest defined in '
+                + `${config.manifest}`);
+
+            // Get the repo binary
+            command.push(`curl -s ${repoDownloadUrl} > /usr/local/bin/repo`);
```

Comment:
All these `curl` and `wget` cmds can be dangerous without wrapper. We don't know if user's image has those pkgs.

Reply:
Right! At the very least however, we should enforce images to have `git` and `python`.

Since `egrep` is also used but in Habitat we can alternatively use `grep -E` https://bldr.habitat.sh/#/pkgs/core/grep/latest